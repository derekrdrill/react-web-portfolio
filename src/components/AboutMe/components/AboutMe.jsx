import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
