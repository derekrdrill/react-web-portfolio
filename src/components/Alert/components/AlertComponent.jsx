import React, { useContext } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = useContext(AlertContext);

  return (
    alert && (
      <StyledAlert severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
