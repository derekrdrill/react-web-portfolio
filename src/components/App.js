import React from 'react';

const AboutMe = React.lazy(() => import('./AboutMe/components/AboutMe'));
const ConnectWithMe = React.lazy(() => import('./ConnectWithMe/components/ConnectWithMe'));
const Homepage = React.lazy(() => import('./Homepage/Homepage'));
const Projects = React.lazy(() => import('./Projects/components/Projects'));

const App = () => (
  <div className='App'>
    <Homepage id='home' />
    <AboutMe id='about-me' />
    <Projects id='projects' />
    <ConnectWithMe id='connect-with-me' />
  </div>
);

export default App;
