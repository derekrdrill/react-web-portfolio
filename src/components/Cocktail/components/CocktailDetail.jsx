import React from 'react';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlayCircle } from '@fortawesome/fontawesome-free-solid';

import { CocktailContext } from '../../Cocktail/context/CocktailContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

const CocktailDetail = () => {
  const { searchResults, searchResultsLength, loading } = React.useContext(CocktailContext);
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      {loading && <LoaderSpinner open />}
      {!loading && searchResults && searchResults.length > 0 && (
        <CocktailDetailContainer container rowSpacing={2}>
          <Grid item xs={1} md={2} />
          <Grid item xs={10}>
            <Typography variant='h5'>
              {`${searchResultsLength} ${searchResultsLength === 1 ? 'drink' : 'drinks'} listed`}
            </Typography>
          </Grid>
          <Grid item xs={12} />
          <Grid item xs={12} />
          {searchResults.map((searchResult, searchResultKey) => (
            <>
              <CocktailResultsContainer key={searchResultKey} container rowSpacing={4}>
                <Grid item xs={1} md={2} />
                <Grid item xs={11} md={3}>
                  <CocktailImage src={searchResult.strDrinkThumb} />
                </Grid>
                <Grid item xs={1} sm={1} />
                <Grid item xs={10} md={5}>
                  <Typography variant='h3'>{searchResult.strDrink}</Typography>
                  <br />
                  <Grid container spacing={4}>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Ingredients</Typography>
                      {Object.keys(searchResult)
                        .filter(result => result.includes('strIngredient'))
                        .map(
                          (ingredient, ingredientKey) =>
                            searchResult[ingredient] && (
                              <Typography
                                key={`${searchResult.idDrink}${ingredientKey}`}
                                variant='subtitle2'
                              >
                                {`${searchResult[`strMeasure${ingredientKey + 1}`] ?? ''} ${
                                  searchResult[ingredient]
                                }`}
                              </Typography>
                            ),
                        )}
                    </Grid>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Glass</Typography>
                      <Typography variant='subtitle2'>{searchResult.strGlass}</Typography>
                    </Grid>
                  </Grid>
                  <br />
                  <Typography variant='h5'>Instructions</Typography>
                  <Typography variant='body2'>{searchResult.strInstructions}</Typography>
                  {searchResult.youtubeData && (
                    <CocktailVideoTutorialLinkContainer item xs={12}>
                      <CocktailVideoTutorialLink
                        href={searchResult.youtubeData.url}
                        target='_blank'
                        rel='noreferrer'
                      >
                        <CocktailVideoTutorial
                          className='youtube-img'
                          src={searchResult.youtubeData.bestThumbnail.url}
                        />
                        <CocktailVideoTutorialPlayButton icon={faPlayCircle} />
                        <CocktailVideoTutorialTitleContainer className='youtube-title'>
                          <CocktailVideoTutorialText variant='body1'>
                            {searchResult.youtubeData.title}
                          </CocktailVideoTutorialText>
                          <CocktailVideoTutorialText variant='subtitle2'>
                            {searchResult.youtubeData.duration}
                          </CocktailVideoTutorialText>
                        </CocktailVideoTutorialTitleContainer>
                      </CocktailVideoTutorialLink>
                    </CocktailVideoTutorialLinkContainer>
                  )}
                </Grid>
                <Grid xs={1} />
                <Grid xs={1} />
                <CocktailResultsDivider
                  item
                  xs={10}
                  darkMode={darkMode}
                  lastItem={searchResultKey + 1 === searchResultsLength}
                />
                <Grid xs={1} />
              </CocktailResultsContainer>
            </>
          ))}
        </CocktailDetailContainer>
      )}
    </>
  );
};

export default CocktailDetail;

export const CocktailDetailContainer = styled(Grid)({
  marginBottom: 150,
  position: 'relative',
  top: 100,
});

export const CocktailImage = styled.img({
  borderRadius: 30,
  height: 275,
  width: 320,
});

export const CocktailResultsContainer = styled(Grid)({
  padding: '40px 0',
});

export const CocktailResultsDivider = styled(Grid)(({ darkMode, lastItem }) => ({
  borderBottom: !lastItem && `solid 1px ${darkMode ? 'gainsboro' : 'grey'}`,
  margin: '0 30px',
  padding: '50px 0',
}));

export const CocktailVideoTutorial = styled.img({
  height: 215,
  width: 315,
});

export const CocktailVideoTutorialLinkContainer = styled(Grid)({
  ':hover': {
    '.youtube-img': {
      opacity: 0.4,
    },
    '.youtube-title': {
      visibility: 'visible',
    },
  },
  height: 215,
  transform: 'translateY(20px)',
  width: 315,
});

export const CocktailVideoTutorialLink = styled.a({
  cursor: 'pointer',
  textDecoration: 'none',
  zIndex: 2,
});

export const CocktailVideoTutorialText = styled(Typography)({
  color: 'white',
  fontWeigt: 'bold',
  zIndex: 0,
});

export const CocktailVideoTutorialTitleContainer = styled.div({
  maxWidth: 300,
  transform: 'translate(10px, -250px)',
  visibility: 'hidden',
});

export const CocktailVideoTutorialPlayButton = styled(FontAwesomeIcon)({
  color: 'red',
  height: 30,
  transform: 'translate(145px, -130px)',
  width: 30,
});
