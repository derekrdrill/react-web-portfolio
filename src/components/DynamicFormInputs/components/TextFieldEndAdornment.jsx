import React from 'react';
import PropTypes from 'prop-types';

import { ShowHideIcon } from '../../ShowHideIcon/ShowHideIcon';

export const handleSetPasswordHidden = (passwordHidden, setPasswordHidden) => {
  setPasswordHidden(!passwordHidden);
  return !passwordHidden;
};

export const TextFieldEndAdornment = ({ passwordHidden, setPasswordHidden, type }) =>
  type === 'password' && (
    <ShowHideIcon
      onClick={
        /* istanbul ignore next */
        () => handleSetPasswordHidden(passwordHidden, setPasswordHidden)
      }
    />
  );

TextFieldEndAdornment.propTypes = {
  passwordHidden: PropTypes.bool,
  setPasswordHidden: PropTypes.func,
  type: PropTypes.string,
};
