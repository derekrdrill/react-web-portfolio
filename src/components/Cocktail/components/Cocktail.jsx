import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';

import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import CocktailDetail from './CocktailDetail';
import CocktailSearch from './CocktailSearch';

const Cocktail = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <CocktailSearch />
      <CocktailDetail />
    </AlertProvider>
  );
};

Cocktail.propTypes = {};

export default Cocktail;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#1B0008' : '#FBAF93',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
