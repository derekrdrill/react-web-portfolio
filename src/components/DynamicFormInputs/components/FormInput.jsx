import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormControl, Grid, InputLabel, MenuItem, Select, TextField } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { TextFieldEndAdornment } from '../components/TextFieldEndAdornment';

export const getVariant = (darkMode, variant) => (darkMode ? 'filled' : variant);
export const getTextFieldType = (passwordHidden, type) =>
  type === 'password' ? (passwordHidden ? 'password' : 'text') : type;

// TODO: condense this all to one style using emotion

// import { css } from '@emotion/react';

export const FormInput = ({ input, form, setForm, value }) => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  const [selectValue, setSelectValue] = useState('');
  const [passwordHidden, setPasswordHidden] = useState(true);

  const selectChange = event => setSelectValue(event.target.value);

  const handleInputChange = (e, setForm) => {
    let id = e.currentTarget.id;
    setForm({ ...form, [id]: e.currentTarget.value });
  };

  return (
    <Grid item xs={input.xs} sm={input.sm} md={input.md}>
      {input.select ? (
        <FormControl className='select' fullWidth>
          <StyledSelectLabel darkMode={darkMode} labelcolor={input.labelColor}>
            {input.label}
          </StyledSelectLabel>
          <StyledSelect
            id={input.id}
            darkMode={darkMode}
            label={input.label}
            value={selectValue}
            variant={getVariant(darkMode, input.variant)}
            fullWidth={input.fullWidth || true}
            onChange={selectChange}
            color={input.color || 'secondary'}
            inputbackgroundcolor={input.inputBackgroundColor}
            inputborderradius={input.inputborderRadius}
            inputpaddingbottom={input.inputPaddingBottom}
            inputfontfamily={input.fontFamily || input.inputFontFamily}
            labelcolor={input.labelColor}
            labelfontfamily={input.fontFamily || input.labelFontFamily}
          >
            {input.selectOptions.map(option => (
              <MenuItem key={option.id} value={option.value}>
                {option.value}
              </MenuItem>
            ))}
          </StyledSelect>
        </FormControl>
      ) : (
        <StyledTextField
          className='text'
          id={input.id}
          darkMode={darkMode}
          label={input.label}
          placeholder={input.placeholder}
          variant={getVariant(darkMode, input.variant)}
          fullWidth={input.fullWidth || true}
          multiline={input.multiline}
          minRows={input.minRows}
          maxRows={input.maxRows}
          onChange={
            /* istanbul ignore next */
            e => handleInputChange(e, setForm)
          }
          value={value}
          size={input.size || 'medium'}
          type={getTextFieldType(passwordHidden, input.type)}
          color={input.color || 'secondary'}
          inputbackgroundcolor={input.inputBackgroundColor}
          inputborderradius={input.inputborderRadius}
          inputpaddingbottom={input.inputPaddingBottom}
          inputfontfamily={input.fontFamily || input.inputFontFamily}
          labelcolor={input.labelColor}
          labelfontfamily={input.fontFamily || input.labelFontFamily}
          InputProps={{
            endAdornment: (
              <TextFieldEndAdornment
                passwordHidden={passwordHidden}
                setPasswordHidden={setPasswordHidden}
                type={input.type}
              />
            ),
          }}
        />
      )}
    </Grid>
  );
};

FormInput.propTypes = {
  form: PropTypes.object,
  input: PropTypes.object,
  setForm: PropTypes.func,
  value: PropTypes.string,
};

// TODO: condense this all to one style using emotion

// const styledInput = css(({
//     inputbackgroundcolor,
//     inputborderradius,
//     inputpaddingbottom,
//     inputfontfamily,
//     labelfontfamily,
//     labelcolor
// }) => ({
//     paddingBottom: inputpaddingbottom,
//     '.MuiInputBase-root': {
//         backgroundColor: inputbackgroundcolor,
//         borderRadius: inputborderradius,
//         fontFamily: inputfontfamily,
//     },
//     '.MuiInputLabel-root': {
//         color: labelcolor,
//         fontFamily: labelfontfamily,
//     },
// }));

export const StyledTextField = styled(TextField)(
  ({
    darkMode,
    inputbackgroundcolor,
    inputborderradius,
    inputpaddingbottom,
    inputfontfamily,
    labelfontfamily,
    labelcolor,
  }) => ({
    paddingBottom: inputpaddingbottom,
    '.MuiInputBase-root': {
      backgroundColor: darkMode ? '#474747' : inputbackgroundcolor,
      borderRadius: inputborderradius,
      color: darkMode ? 'beige' : 'inherit',
      fontFamily: inputfontfamily,
    },
    '.MuiInputLabel-root': {
      color: darkMode ? 'white' : labelcolor,
      fontFamily: labelfontfamily,
    },
  }),
);

export const StyledSelectLabel = styled(InputLabel)(({ darkMode, labelcolor }) => ({
  color: darkMode ? 'white' : labelcolor,
}));

export const StyledSelect = styled(Select)(
  ({
    darkMode,
    inputbackgroundcolor,
    inputborderradius,
    inputpaddingbottom,
    inputfontfamily,
    labelfontfamily,
    labelcolor,
  }) => ({
    paddingBottom: inputpaddingbottom,
    '&.MuiInputBase-root': {
      backgroundColor: darkMode ? '#474747' : inputbackgroundcolor,
      borderRadius: inputborderradius,
      color: darkMode ? 'beige' : 'inherit',
      fontFamily: inputfontfamily,
    },
    '.MuiInputLabel-root': {
      color: darkMode ? 'beige' : labelcolor,
      fontFamily: labelfontfamily,
    },
  }),
);
