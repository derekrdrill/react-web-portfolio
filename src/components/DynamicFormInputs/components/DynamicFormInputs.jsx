import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@mui/material';

import { FormInput } from './FormInput';

export const DynamicFormInputs = ({ inputs, form, setForm }) => (
  <Grid container spacing={2}>
    {inputs.map(input => {
      let value = form[input.id];

      return <FormInput key={input.id} form={form} input={input} setForm={setForm} value={value} />;
    })}
  </Grid>
);

DynamicFormInputs.propTypes = {
  inputs: PropTypes.array,
  form: PropTypes.object,
  setForm: PropTypes.func,
};
