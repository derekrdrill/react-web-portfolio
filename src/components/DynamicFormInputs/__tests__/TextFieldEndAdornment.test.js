import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { TextFieldEndAdornment } from '../components/TextFieldEndAdornment';

import { handleSetPasswordHidden } from '../components/TextFieldEndAdornment';

const renderer = new ShallowRenderer();

describe('TextFieldEndAdornment tests', () => {
  it('renders correctly', () => {
    renderer.render(
      <TextFieldEndAdornment passwordHidden setPasswordHidden={jest.fn()} type='password' />,
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('runs handleSetPasswordHidden', () => {
    expect(handleSetPasswordHidden(true, jest.fn())).toEqual(false);
    expect(handleSetPasswordHidden(false, jest.fn())).toEqual(true);
  });
});
