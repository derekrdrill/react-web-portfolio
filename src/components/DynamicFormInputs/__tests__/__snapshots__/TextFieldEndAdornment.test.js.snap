// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldEndAdornment tests renders correctly 1`] = `
<ShowHideIcon
  onClick={[Function]}
/>
`;
